Analysis/Tasks/PWGCF/FemtoDream/include/FemtoDream/FemtoDreamMath.h

```diff
@@ -0,0 +1,127 @@
+  {
+    ROOT::Math::PtEtaPhiMVector vecpart1(part1.pt(), part1.eta(), part1.phi(), mass1);
+    ROOT::Math::PtEtaPhiMVector vecpart2(part2.pt(), part2.eta(), part2.phi(), mass2);
+    return getkstar(vecpart1, vecpart2);
+  }
+
+  static float getkT(const ROOT::Math::PtEtaPhiMVector& part1, const ROOT::Math::PtEtaPhiMVector& part2);
+
+  template <typename T>
+  static float getkT(const T& part1, const float mass1, const T& part2, const float mass2)
+  {
+    ROOT::Math::PtEtaPhiMVector vecpart1(part1.pt(), part1.eta(), part1.phi(), mass1);
+    ROOT::Math::PtEtaPhiMVector vecpart2(part2.pt(), part2.eta(), part2.phi(), mass2);
+    return getkT(vecpart1, vecpart2);
+  }
+
+  static float getmT(const ROOT::Math::PtEtaPhiMVector& part1, const ROOT::Math::PtEtaPhiMVector& part2);
+
+  template <typename T>
+  static float getmT(const T& part1, const float mass1, const T& part2, const float mass2)
+  {
+    ROOT::Math::PtEtaPhiMVector vecpart1(part1.pt(), part1.eta(), part1.phi(), mass1);
+    ROOT::Math::PtEtaPhiMVector vecpart2(part2.pt(), part2.eta(), part2.phi(), mass2);
+    return getmT(vecpart1, vecpart2);
+  }
+
+ protected:
+
+  float mMassOne = 0.f;
+  float mMassTwo = 0.f;
+};
+
+inline float FemtoDreamMath::getkstar(const ROOT::Math::PtEtaPhiMVector& part1, const ROOT::Math::PtEtaPhiMVector& part2)
+{
+  ROOT::Math::PtEtaPhiMVector trackSum = part1 + part2;
+
+  float beta = trackSum.Beta();
+  float betax = beta * std::cos(trackSum.Phi()) * std::sin(trackSum.Theta());
+  float betay = beta * std::sin(trackSum.Phi()) * std::sin(trackSum.Theta());
+  float betaz = beta * std::cos(trackSum.Theta());
+
+  float px1 = part1.pt() * std::cos(part1.Phi());
+  float py1 = part1.pt() * std::sin(part1.Phi());
+  float pz1 = part1.pt() * std::sinh(part1.Eta());
+  float px2 = part2.pt() * std::cos(part2.Phi());
+  float py2 = part2.pt() * std::sin(part2.Phi());
+  float pz2 = part2.pt() * std::sinh(part2.Eta());
+
+  ROOT::Math::PxPyPzMVector PartOneCMS(px1, py1, pz1, part1.M());
+  ROOT::Math::PxPyPzMVector PartTwoCMS(px2, py2, pz2, part2.M());
+
+  ROOT::Math::Boost boostPRF = ROOT::Math::Boost(-betax, -betay, -betaz);
+  PartOneCMS = boostPRF(PartOneCMS);
+  PartTwoCMS = boostPRF(PartTwoCMS);
+
+  ROOT::Math::PxPyPzMVector trackRelK = PartOneCMS - PartTwoCMS;
+  return 0.5 * trackRelK.P();
+}
+
+inline float FemtoDreamMath::getkT(const ROOT::Math::PtEtaPhiMVector& part1, const ROOT::Math::PtEtaPhiMVector& part2)
+{
+  ROOT::Math::PtEtaPhiMVector trackSum = part1 + part2;
+  float trackRelkT = 0.;
+  trackRelkT = 0.5 * trackSum.Pt();
+  return trackRelkT;
+}
+
+inline float FemtoDreamMath::getmT(const ROOT::Math::PtEtaPhiMVector& part1, const ROOT::Math::PtEtaPhiMVector& part2)
+{
+  ROOT::Math::PtEtaPhiMVector trackSum = part1 + part2;
+  float relmT = 0.;
+  float trackRelkT = 0.5 * trackSum.Pt();
+  float averageMass = 0.5 * (part1.M() + part2.M());
+  relmT = TMath::Sqrt(pow(trackRelkT, 2) + pow(averageMass, 2));
+  return relmT;
```

Comment:
in order to avoid redundancy please call here getkT for the computation of kT
and then also no further assignments
{ 
  getmT( part1, part2, mass1, mass2) {  return std::sqrt(std::pow(getkT(..), 2)  + pow(0.5 * (m1+m2), 2)); 
}